analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/LocksReleasedAllPaths.Mutex.cs

```diff
@@ -94,7 +94,7 @@ public void Noncompliant(Mutex paramMutex, Mutex paramMutex2, Foo foo)
     public void NoncompliantReleasedThenAcquiredAndReleased(Mutex paramMutex)
     {
         paramMutex.ReleaseMutex();
-        paramMutex.WaitOne(); // FN, after this acquire it's not released on all paths
+        paramMutex.WaitOne(); // Noncompliant
```

Comment:
There're some FNs that looks like they should be `Noncompliant` now:
Line 16
Line 19
Line 22
Line 26
Line 55

Reply:
Don't invest too much time in this investigation. It might be related (and fixed) by #5370

Reply:
Yes, the problem seems to be related to that. From what I can see, in these cases, for all the explored branches we evaluate the "remove lock" method.